### DISCUSSION TITLE
.net 6 System.IO.FileNotFoundException "System.Data.SqlClient"

### DISCUSSION THREAD

**Discussion by cli00004** [JasperFx/marten#2056] [Q&A]
Hi, I'm getting the following exception using Marten 4.3.1 in .NET 6. Running in docker kubernetes linux 
![image](https://user-images.githubusercontent.com/94087985/150745188-38032fe7-a76d-4fce-b82f-7691b7f33245.png)

**Comment by jeremydmiller** [comment]
@cli00004 I really don't think this has anything to do with Marten, unless there's a Nuget diamond dependency problem somewhere. Marten has no dependency on Sql Server.

**Comment by cli00004** [comment]
Thanks @jeremydmiller  for replying. That is what I thought, I don't see Marten is referencing sqlclient, but the thing is when I try connecting with raw Npgsql command, it works fine, but when I use Marten, it gives me this error at runtime, hmm

**Comment by cli00004** [comment]
After a bit more testing, and what i found is the following works with raw sql query. I'm running a very simple asp.net 6 app without any dependency except Marten in Linux docker/k8s environment 
```
using var session = _documentStore.QuerySession();
var list = session.Query<Test>("select data from public.mt_doc_test").FirstOrDefault();
```

But this doesn't, which I assume they should have been the same thing? It seems like there some sql generation tool issue
```
using var session = _documentStore.QuerySession();
var list = session.Query<Test>().FirstOrDefault();
```

**Reply by jeremydmiller** [reply]
No, that's not the *exact* same functionality. Can you post your actual exception? Without more details, I can't even make a guess here.

**Reply by jeremydmiller** [reply]
Doh, got enough coffee. The dynamic codegen stuff in Marten has to walk up your assembly's dependencies and reference each dependency. *Somewhere*, you have a transitive dependency on the missing Nuget. You might hunt that down, and possibly add that explicitly to your main application to get around this. It's a Nuget problem one way or another.

**Reply by cli00004** [reply]
There is only one application and it's directly referencing Marten like this
![image](https://user-images.githubusercontent.com/94087985/151103276-5526a306-7b4e-4282-81b3-ecff30c69057.png)

And here's the exception, basically to do with 2 things, System.Data.Common and System.Data.SqlClient
![image](https://user-images.githubusercontent.com/94087985/151040856-defb1079-3729-4d9c-9873-79410389259a.png)

**Reply by cli00004** [reply]
There's defintely issue with the latest Marten, I have downgraded it to our previous working version, 2.7.1, it works now. It might be related to the Weasle thing
